src/backend/auth.api.ts

```diff
@@ -143,18 +143,20 @@ const getUserByUserId = async (userId: string) => {
 };
 
 const login = async (email: string, password: string) => {
+  // Delete current session, if any
   try {
```

Comment:
Does having to try blocks handle all the issues of login? and have you tested it that does the second try initiates after the execution of 1st try-catch block

Reply:
The `deleteSession()` from appwrite returns error: `generate_unauthorized_scope` in case there is no active session.

In the previous flow, `deleteSession()` and `createEmailSession()` were in the same try-catch block. Thus, the catch block was catching the error thrown by `deleteSession()` which was just an indication that there is no active session and could've been ignored. Meaning, in the previous flow, the `createEmailSession()` was never invoked.

By adding a separate try-catch block for `deleteSession()`, the previously thrown error for no active session is ignored and rest errors are caught.

Next, the second try-catch block is where the actual session is being created and the user details are fetched. Any errors while creating a new session are caught and logged.

**To summarize**

- While deleting an existing session, if there is an error apart from no active session, then the user will get an error message because only one session per browser is allowed by appwrite. I found no errors so to speak while deleting an active session.
- If active session is successfully deleted, we proceed to create a session and fetch user details. If there is an error in this step, user is again shown an error message. Mostly, invalid login credentials.
- If session is created and user details are fetched successfully, then the user is logged in.

Yes, the control does reach the second try-catch and user is logged in without any issues. I've tested the login flow many times and have found no issues.